packages/playwright-core/src/remote/playwrightServer.ts

```diff
@@ -41,16 +40,17 @@ type ServerOptions = {
 };
 
 export class PlaywrightServer {
-  private _preLaunchedPlaywright: Playwright | undefined;
+  private _playwright: Playwright;
   private _options: ServerOptions;
   private _wsServer: WSServer;
 
   constructor(options: ServerOptions) {
     this._options = options;
     if (options.preLaunchedBrowser)
-      this._preLaunchedPlaywright = options.preLaunchedBrowser.attribution.playwright;
+      this._playwright = options.preLaunchedBrowser.attribution.playwright;
     if (options.preLaunchedAndroidDevice)
-      this._preLaunchedPlaywright = options.preLaunchedAndroidDevice._android.attribution.playwright;
+      this._playwright = options.preLaunchedAndroidDevice._android.attribution.playwright;
+    this._playwright ??= createPlaywright({ sdkLanguage: 'javascript', isServer: true });
```

Comment:
It looks like both `sdkLanguage` and `socksProxyPort` options should become per-browser instead of per-playwright.

Reply:
Per-browser, or maybe per connection instead?